packages/opentelemetry-exporter-collector-grpc/src/util.ts

```diff
@@ -105,6 +104,10 @@ export function send<ExportItem, ServiceRequest>(
   }
 }
 
-function removeProtocol(url: string): string {
-  return url.replace(/^https?:\/\//, '');
+export function fixUrl(url: string): string {
+  const serverAddress = url.replace(/^(http|grpc)s?:\/\//, '');
```

Comment:
Maybe we should use the `url` module to extract only the host and port part of the string?

```typescript
const myURL = new URL('https://example.org:81/foo');
console.log(myURL.host);
// Prints example.org:81

// pathname is `/foo` 
if (myURL.pathname !== '/') {
  console.log("URL path set")
}
```

Reply:
that sure would be a lot cleaner/more consistent, heh.

Reply:
be aware that `new URL()` may throw if the url is not parseable. Not sure if such an exception should escape here to transferred into some log.

Reply:
better to throw the exception than to silently fail with a pile of DNS lookup errors because the hostname is nonsense heh

Reply:
is there a preference on approach here? @Flarna @dyladan

Reply:
As far as I'm aware we are supposed to do our best to never throw when possible. I know this is very important in the API, but in the SDK configuration it is less of an issue. I would lean towards allowing this to throw if the URL is unparseable.

Reply:
Maybe other @open-telemetry/javascript-maintainers or @open-telemetry/technical-committee has an opinion here

Reply:
I would be fine with throwing if url is wrong as it will not work anyway. with regards to regex or URL, both will be working fine I guess, but URL might be indeed cleaner and might cover cases we are not aware of now.

Reply:
Two rules from https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/error-handling.md:
>The API or SDK may fail fast and cause the application to fail on initialization, e.g. because of a bad user config or environment, but MUST NOT cause the application to fail later at run time, e.g. due to dynamic config settings received from the Collector.
>The SDK MUST NOT throw unhandled exceptions for errors in their own operations. For example, an exporter should not throw an exception when it cannot reach the endpoint to which it sends telemetry data.

Both rules arguably are applicable here (is this a "cannot reach" case?). Can we fail fast when URL is wrong? If yes then that's a valid choice (but we don't have to, it is only a MAY clause).

If we are going to discover the URL is wrong at a later time (not startup) then definitely don't fail, just log it (or whatever equivalent we do in the js world).

Reply:
I missed the first rule when I read the spec thanks for the heads up. Looks like failing here is totally acceptable.

Reply:
@tigrannajaryan I think in this case we can assume user added a link which looks like for example `foobar`. This is not a valid link and will never be so it could fail fast and using `URL` object will give us such benefit. This is not about that we want to fail whole app because suddenly the correct url cannot reach the endpoint but the url is simply wrong .
`removeProtocol` is being called during initialisation so the failure will be immediate not after some time when first span will be shown. This way I think it would be essential information for the user, instead of failing silently. WDYT ?

Reply:
@obecny yes, I believe this is a case when we _may_ fail. We don't have to but we are allowed to.

If we want to go a step further it may make sense to see what other languages are doing in the same situation and propose an amendment to the error handling doc I linked to so that the spec is less ambiguous and the behaviour of different languages is uniform.

Reply:
> is there a preference on approach here? @Flarna @dyladan

I'm fine with both variants. Main target was to point out that `new URL` is different in this compared to all other variants (`url.parse()`, regex, ...) which often results in surprises.

Reply:
unresolving this thread for @obecny's concern about throwing vs warning